driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -120,31 +124,33 @@ public void run() {
 
         Executor initExecutor = manager.cluster.manager.configuration.getPoolingOptions().getInitializationExecutor();
 
-        ListenableFuture<List<Void>> allConnectionsFuture = Futures.allAsList(connectionFutures);
+        ListenableFuture<List<Void>> allConnectionsFuture = Futures.successfulAsList(connectionFutures);
 
         final SettableFuture<Void> initFuture = SettableFuture.create();
-        Futures.addCallback(allConnectionsFuture, new FutureCallback<List<Void>>() {
+        Futures.addCallback(allConnectionsFuture, new MoreFutures.SuccessCallback<List<Void>>() {
             @Override
             public void onSuccess(List<Void> l) {
+                // Some of the connections might have failed, keep only the successful ones
+                ListIterator<Connection> it = connections.listIterator();
```

Comment:
Yes, ClusterNameMismatch/UnsupportedProtocolVersion should be bubbled up to `SessionManager#maybeAddPool`, which will handle them accordingly (mark the node down and don't trigger a reconnect).
I'm fixing this on my local integration branch because it's easier to do on top of 821/822 (very similar to handling a 'wrong keyspace' error).
Can you keep a note to retry these tests with 822?